### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: lib/algorithms/Ed25519.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bs58 = require('bs58');
+const chloride = require('chloride');
+
+module.exports = class Algorithms {
+  constructor() {
+  }
```
Comment (davidlehn) [inline lib/algorithms/Ed25519.js]: Could omit the empty constructor in all algorithms classes.

File: lib/algorithms/Ed25519.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bs58 = require('bs58');
+const chloride = require('chloride');
+
+module.exports = class Algorithms {
```
[the remark this answers is not in the document]
Comment (davidlehn) [reply lib/algorithms/Ed25519.js]: Could perhaps omit the name (`module.exports = class { ... }`) but might be useful for debugging?

File: lib/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+const Rsa = require('./algorithms/Rsa');
+
+const api = {};
+module.exports = api;
+
+api.sign = (
+  {algorithm, hashType, plaintext, privateKeyBase58, privateKeyPem, sharedKey},
+  callback) => {
+  const signer = api.use(algorithm);
+  async function fn() {
```
[the remark this answers is not in the document]
Comment (mattcollier) [reply lib/index.js]: In looking at implementing a helper, seems to me there are good benefits to a shared scope here.  Have an idea that does not involve passing all the parameters again, or am I misunderstanding your suggestion?

[the remark this answers is not in the document]
Comment (dlongley) [reply lib/index.js]: This might work:

https://gist.github.com/dlongley/dcc7db43270ae53a90a5c89604e58b19

[the remark this answers is not in the document]
Comment (dlongley) [reply lib/index.js]: It also seems like you could have `use` throw that unknown algorithm error and remove that too.

**Pull request merged**